tygron-environment/src/main/java/nl/tudelft/contextproject/tygron/api/Environment.java

```diff
@@ -324,23 +330,38 @@ public BuildRequest(Stakeholder stakeholder, Function function, int floors, Poly
    */
   private Polygon getSuitableLand(Polygon availableLand, int surface) {
     getMapWidth();
-    double margin = 0.5 * mapWidth;
-    double square = margin;
+    Random random = new Random();
+    Polygon selectedLand;
+    Polygon intersection;
     
-    Polygon selectedLand = PolygonUtil.polygonIntersection(createPolygon(square), availableLand);
-    while (withinMap(square) && !withinMargin(selectedLand, surface)) {
-      margin = 0.5 * margin;
-      if (selectedLand.calculateArea2D() < surface) {
-        square = square + margin;
+    double x1;
+    double y1;
+    double x2;
+    double y2;
+    // Select a random rectangle in the map and get the part of land that is available.
+    do {
+      x1 = random.nextDouble() * mapWidth;
+      y1 = random.nextDouble() * mapWidth;
+      x2 = random.nextDouble() * mapWidth;
+      y2 = random.nextDouble() * mapWidth;
+      selectedLand = PolygonUtil.makeRectangle(x1, y1, x2, y2);
+      intersection = PolygonUtil.polygonIntersection(selectedLand, availableLand);
+    } while (intersection.calculateArea2D() < surface);
```

Comment:
Yes, because this method is not called unless there is plenty of land available. The coordinates are randomised, therefore a large area will likely be selected which contains the required surface. This was even tested on small corners of the map and it always finds a large enough chunk of land.